ec-gpu/src/lib.rs

```diff
@@ -1,22 +1,43 @@
-/// Describes how to generate the elliptic curve operations for
-/// - `Scalar`
-/// - `Fp`
-/// - `Fp2`
-/// - `G1`
-/// - `G2`
-pub trait GpuEngine {
-    type Scalar: GpuField;
-    type Fp: GpuField;
+/// The name that is used in the GPU source code to identify the item that is used.
+///
+/// It is automatically implemented for the [`GpuField`] trait.
+pub trait GpuName {
```

Comment:
why make it its own trait and not part pf GpuField?

Reply:
See https://github.com/filecoin-project/ec-gpu/pull/30#discussion_r937057634, it's also used for things that are not fields.